qiskit/optimization/algorithms/optimization_algorithm.py

```diff
@@ -22,6 +22,19 @@
 from ..problems.quadratic_program import QuadraticProgram, Variable
 
 
+class OptimizationResultStatus(Enum):
+    """Termination status of an optimization algorithm."""
+
+    SUCCESS = 0
+    """the optimization algorithm succeeded to find an optimal solution."""
+
+    FAILURE = 1
```

Comment:
Is FAILURE ever used anywhere - I could not find it used but maybe its there in case its needed for the future. I assume if an OptimizationResult is created with such a status then maybe x is set to an empty array to comply with the interface (None arguably is not valid). Either way with status of FAILURE the result object would not really be expected to be usable.

Reply:
Now it is not used, this is true. Do we need this? Well, maybe. I agree, if status is failure then `x`, `fval` don't make much sense.